### PR TITLE
NServiceBus.Extensions.Hosting doco

### PR DESCRIPTION
**Pull request by SeanFeldman**
- Component
- Doco

_Replaces PR https://github.com/Particular/docs.particular.net/pull/4579_

### CODE REVIEW

File: components/components.yaml [hunk 1]
```diff
@@ -823,3 +823,12 @@
   ProjectUrl: https://github.com/ParticularLabs/NServiceBus.AwsLambda
   NugetOrder:
     - NServiceBus.AwsLambda.SQS
+
+- Key: Microsoft.Hosting
+  Name: Microsoft.Hosting
+  DocoUrl: nservicebus/hosting/microsoft-extensions
```
Comment (timbussmann) [inline components/components.yaml]: I think this shouldn't be on the `extensions-hosting` URL as there are many Microsoft.Extensions.* packages. How about `extensions-hosting`?

File: components/components.yaml [hunk 2]
```diff
@@ -823,3 +823,12 @@
   ProjectUrl: https://github.com/ParticularLabs/NServiceBus.AwsLambda
   NugetOrder:
     - NServiceBus.AwsLambda.SQS
+
+- Key: Microsoft.Hosting
```
Comment (timbussmann) [inline components/components.yaml]: should this be Extensions.Hosting?

File: components/components.yaml [hunk 3]
```diff
@@ -823,3 +823,12 @@
...
+  Name: Microsoft.Hosting
```
Comment (timbussmann) [inline components/components.yaml]: should this be NServiceBus.Extensions.Hosting?

File: components/components.yaml [hunk 4]
```diff
@@ -823,3 +823,12 @@
...
+  DocoUrl: nservicebus/hosting/microsoft-extensions
+  GitHubOwner: Particular
+  Category: DependencyInjection
```
Comment (timbussmann) [inline components/components.yaml]: should this be `Host`?

Comment (SeanFeldman) [reply components/components.yaml]: Missed it, yes

File: Snippets/Extensions.Hosting/Extensions.Hosting_1/Configuration.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        async Task ConfigureHost()
+        {
+            #region extensions-host-configuration
+            var host = new HostBuilder()
+                .ConfigureServices(serviceCollection =>
+                {
+                    var endpointConfiguration = new EndpointConfiguration("MyEndpoint");
+                    // configure endpoint here
+
+                    serviceCollection.AddNServiceBus(endpointConfiguration);
```
Comment (SeanFeldman) [inline Snippets/Extensions.Hosting/Extensions.Hosting_1/Configuration.cs]: I'm assuming we'll change this to `host.UseNServiceBus(...)` once https://github.com/Particular/NServiceBus.Extensions.Hosting/pull/13 is merged.

**Comment by SeanFeldman** [comment]
Marking as WIP as we'll need to change the sample in this PR to use the new API added in the recent PR https://github.com/Particular/NServiceBus.Extensions.Hosting/pull/13.

File: Snippets/Extensions.Hosting/Extensions.Hosting_1/Configuration.cs (hunk 5 above)
Comment (SeanFeldman) [reply Snippets/Extensions.Hosting/Extensions.Hosting_1/Configuration.cs]: Nevermind. This snippet will stay as-is for the WebHost. We'll add another one for the GenericHost.

**Comment by SeanFeldman** [comment]
Split the snippet into two: WebHost and GenericHost.

**Pull request merged**